engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
The PR was created on the basis that getNodeId() returned null and that is why nodeId was cached in the AllocatedContainer. If allocatedContainer is null on line 1014, what is the point of caching nodeId (and host)? Is the root cause not clear or are we adding a bunch of defensive fixes in this PR?